includes/DataObjects/User.php

```diff
@@ -644,7 +644,7 @@ public function isAdmin()
     
 	public function isCheckuser()
 	{
-		return $this->checkuser == 1 || $this->oauthCanCheckUser();
+		return $this->checkuser == 1 || ($this->isOAuthLinked() && $this->oauthCanCheckUser());
```

Comment:
Maybe a better approach would be to add a isOAuthLinked() check in the actual methods themselves (e.g., oauthCanCheckUser()) instead of hunting around looking for calls to the methods and short-circuiting there?